### PR TITLE
Contact+about

### LINKED ISSUE #22: Basic structure for Contact and About pages
This might include allowing for the input from Kurt for his about content.

### PR DESCRIPTION
**Pull request by kmdunn5**
closes #22 

Styles aren't a thing really right now. We can come back to this later. Functionality should be good.

### CODE REVIEW

**Comment by kmdunn5** [comment]
@ChuckBTaylor because of the `{...props}` that was working I get this error now that I'm passing in a function.

Screenshot 2024-01-21 at 1 44 52 PM

**Comment by kmdunn5** [comment]
This PR also has some env variables. We can discuss how to share these without exposing them.

File: src/common/constants/constants.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import { Genre, Role } from './enums'
+import { Genre, Input, Role } from "./enums";
+import { TextFieldProps } from "@mui/material";
```
Comment (ChuckBTaylor) [inline src/common/constants/constants.ts:2]: 💭 Not a huge fan of importing MUI to the constants folder.  I see why you're doing it, but I wonder if there's a different way.

File: src/components/atoms/buttons/PrimaryButton.tsx
Comment (ChuckBTaylor) [inline src/components/atoms/buttons/PrimaryButton.tsx:1]: This feels unnecessary

File: src/components/atoms/formInput/TextInput.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { TextField, TextFieldProps } from "@mui/material";
+import { Input } from "../../../common/constants/enums";
+import { kurtTheme } from "../../../theme";
+import { inputToFormProps } from "../../../common/constants/constants";
+
+export type TextInputProps = {
+  fieldName: Input
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
```
Comment (ChuckBTaylor) [inline src/components/atoms/formInput/TextInput.tsx:8]: ❓ Is this just to make the `onChange` prop of `TextFieldProps` required?

File: src/components/molecules/About/AboutPage.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  const navigate = useNavigate()
+
+  const title = Kurt.title
+  const description = Kurt.description
+  
+  return (
+    <>
+      <Typography variant="h3">{title}</Typography>
+      <Typography>{description}</Typography>
+      {Kurt.photos.map((photo) => <img src={photo} alt="Kurt Sensenbrenner flying an airplane"/>)}
```
Comment (ChuckBTaylor) [inline src/components/molecules/About/AboutPage.tsx:15]: The `alt` could be part of the object that define which photo we're using.

```typescript
Kurt = {
  ...,
  photos: [
    {
      url: "",
      alt: ""
    }
  ]
}
```

File: src/components/molecules/About/AboutPage.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { Button, Typography } from "@mui/material";
+import { useNavigate } from "react-router-dom";
+import { Kurt } from "../../../assets/personalInfo/personalInfo"
+
+export function AboutPage (){
+  const navigate = useNavigate()
```
Comment (ChuckBTaylor) [inline src/components/molecules/About/AboutPage.tsx:6]: 👍 I feel like we should use this everywhere.

File: src/components/molecules/About/AboutPage.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+  const title = Kurt.title
+  const description = Kurt.description
+  
+  return (
+    <>
+      <Typography variant="h3">{title}</Typography>
+      <Typography>{description}</Typography>
```
Comment (ChuckBTaylor) [inline src/components/molecules/About/AboutPage.tsx:14]: What does `<Typography>` give us w/o a variant?

File: src/components/molecules/contactForm/ContactForm.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  const form = useRef();
+
+  const sendEmail = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const currentForm = form.current
+
+    // TODO: enValid package
```
Comment (ChuckBTaylor) [inline src/components/molecules/contactForm/ContactForm.tsx:33]: 💭 We probably don't need a package for this. All it's reall doing is accessing the env variables and making them available in an exported object.

File: src/components/molecules/contactForm/ContactForm.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    if (currentForm == null || !process.env.REACT_APP_EMAIL_SERVICE_ID || !process.env.REACT_APP_EMAIL_TEMPLATE_ID || !process.env.REACT_APP_EMAIL_PUBLIC_ID) return // don't love this, but otherwise my sendForm() errors
+
+    emailjs.sendForm(process.env.REACT_APP_EMAIL_SERVICE_ID, process.env.REACT_APP_EMAIL_TEMPLATE_ID, currentForm, process.env.REACT_APP_EMAIL_PUBLIC_ID)
+      .then(() => {
+        setFormSubmitted(true);
+      }, (error) => {
+          console.log(error.text); // We should figure out what we want to show for "Your email didn't send"
```
[the remark this answers is not in the document]
Comment (ChuckBTaylor) [reply src/components/molecules/contactForm/ContactForm.tsx:40]: I've asked Vika to come up with something

File: src/components/atoms/formInput/TextInput.tsx (hunk 2 above)
Comment (kmdunn5) [reply src/components/atoms/formInput/TextInput.tsx:8]: Good point, kind of was to start. But I feel like we probably don't need it to be required.

File: src/components/molecules/About/AboutPage.tsx (hunk 4 above)
Comment (kmdunn5) [reply src/components/molecules/About/AboutPage.tsx:6]: Navigate? Yes. Kind of the whole point of react navigation. Use the hook it gives you.

File: src/components/molecules/contactForm/ContactForm.tsx (hunk 6 above)
Comment (kmdunn5) [reply src/components/molecules/contactForm/ContactForm.tsx:33]: Yeah, it's why I didn't worry about it here. We can add it if we want later, but not necessary now.

File: src/components/atoms/formInput/TextInput.tsx (hunk 2 above)
Comment (ChuckBTaylor) [reply src/components/atoms/formInput/TextInput.tsx:8]: https://github.com/mwkwsd/sensensomething/pull/56/files

Comment (kmdunn5) [reply src/components/atoms/formInput/TextInput.tsx:8]: Cool. Definitely will want to walk through that at some point. `[P in K]-?: T[P]` looks like literally nothing to me.

File: src/components/molecules/About/AboutPage.tsx (hunk 5 above)
Comment (kmdunn5) [reply src/components/molecules/About/AboutPage.tsx:14]: just a regular `<p>` tag

File: src/components/atoms/formInput/TextInput.tsx (hunk 2 above)
Comment (ChuckBTaylor) [reply src/components/atoms/formInput/TextInput.tsx:8]: That's the idea.
Or... wait
do you mean it looks like it does nothing
Or do you mean you don't understand it?

Comment (mwilkins74) [reply src/components/atoms/formInput/TextInput.tsx:8]: Agreed with Ken. 
I'd love an explanation of that code. I've seen generics before but this is totally new code to me as well.

Comment (kmdunn5) [reply src/components/atoms/formInput/TextInput.tsx:8]: I mean I don't understand it.

Comment (ChuckBTaylor) [reply src/components/atoms/formInput/TextInput.tsx:8]: I started to write something up about this. Then it got too much. Happy to go over it in detail if you want. Basically, this util type turns the keys in T provided in K required on T

**Review by ChuckBTaylor — APPROVED** [review]

**Pull request merged**